chore: update README for new reasons (#182)
ariga/atlas-operator: README.md

| ------ | ----------- |
| Reconciling | The operator is reconciling the desired state with the actual state of the database |
| ReadSchema | There was an error about reading the schema from ConfigMap or database credentials |

> | GettingDevDB | failed to get the devdb resource, in case we are using the devdb for nomalization |

Author: ```suggestion
| GettingDevDB | Failed to get a [Dev Database](https://atlasgo.io/concepts/dev-database) in case we are using the devdb for nomalization |
```

| Reason | Description |
| ------ | ----------- |
| Reconciling | The operator is reconciling the desired state with the actual state of the database |

> | GettingDevDB | failed to get the devdb resource, in case we are using the devdb for compute plan for migration |

Reviewer: ```suggestion
| GettingDevDB | Failed to get a [Dev Database](https://atlasgo.io/concepts/dev-database) which is required to compute a migration plan |
```

| ReadingMigrationData | failed to read the migration directory from `ConfigMap`, Atlas Cloud or invalid database credentials |
| ProtectedFlowError | occurred when the migration is protected and the operator is not able to apply it |

> | ApprovalPending | The migration is protected and waiting for approval on Atlas Cloud |

Reviewer: ```suggestion
| ApprovalPending | Applying the migration requires manual approval on Atlas Cloud |
```
